### PR TITLE
[RHCLOUD-19793] Source handler tests update

### PR DESCRIPTION
**Pull request by petracihalova**
Tests update for source handlers - part I. 
- main goal is to add/update tests to better tenancy testing
- each commit contains one updated or added test (or new helper function)
- changes for other functions in source handler will be part of next PR

handlers covered:
 - SourceListAuthentications()
 - SourceTypeListSource()
 - ApplicationTypeListSource()

**JIRA:** [RHCLOUD-19793](https://issues.redhat.com/browse/RHCLOUD-19793)

### CODE REVIEW

File: source_handlers_test.go [hunk 1]
```diff
@@ -205,16 +330,28 @@ func TestSourceTypeSourceSubcollectionList(t *testing.T) {
...
+	}
+
+	sourcesBelongToTenant, err := checkAllSourcesBelongToTenant(tenantId, out.Data)
+	if err != nil {
+		t.Error(err)
+	}
 
+	if !sourcesBelongToTenant {
+		t.Error("the tenant doesn't own all sources")
 	}
```
Comment (MikelAlejoBR) [inline source_handlers_test.go]: Maybe you don't need a boolean value being returned from the `checkAllSourcesBelongToTenant` function. You could simply make it return `err` if they don't belong, and you just have to keep the `if err != nil` check. What do you think?

**Review by MikelAlejoBR — APPROVED** [review]
LGTM! Just a minor comment.

File: source_handlers_test.go (hunk 1 above)
Comment (petracihalova) [reply source_handlers_test.go]: good point, thanks ... solved in new commit

**Pull request merged**